src/data/rat/cast.lean

```diff
@@ -277,3 +277,45 @@ end
 
 instance rat.subsingleton_ring_hom {R : Type*} [semiring R] : subsingleton (ℚ →+* R) :=
 ⟨ring_hom.ext_rat⟩
+
+/-- Integer values of a morphism `φ` completely determine `φ`. -/
+theorem ext_hom_int {α} [group_with_zero α] (φ₁ φ₂ : monoid_with_zero_hom ℚ α)
+  (same_on_int : ∀ k : ℤ, φ₁ k = φ₂ k) : φ₁ = φ₂ :=
```

Comment:
As mentioned on zulip, this lemma becomes more powerful when stated as follows:
```suggestion
  (same_on_int : φ₁.comp (int.cast_ring_hom.to_monoid_with_zero_hom) = φ₂.comp (int.cast_ring_hom.to_monoid_with_zero_hom)) : φ₁ = φ₂ :=
```

There's a library note on "partially-applied ext lemmas" that mentions this.

Reply:
While we're at it, should we also perform a similar change on `ext_hom_int` : 
```lean
theorem ext_hom_pnat' {α} [group_with_zero α] {φ₁ φ₂ : monoid_with_zero_hom ℤ α}
  (same_on_neg_one : φ₁ (-1) = φ₂ (-1))
  (same_on_pnat : φ₁.comp int.of_nat_hom.to_monoid_with_zero_hom =
    φ₂.comp int.of_nat_hom.to_monoid_with_zero_hom) : φ₁ = φ₂
```
I'm unsure whether this is a good idea or not, because
 * This lemma has two assumption instead of one
 * I don't know if I should state this with `int.of_nat_hom` or with `nat.cast_add_monoid_hom ℤ`

Reply:
> I don't know if I should state this with int.of_nat_hom or with nat.cast_add_monoid_hom ℤ

It shouldn't really matter as long as both simplify to the same thing. That might be a good idea though, because that lets you chain this lemma with one that says "if two `monoid_with_zero_hom`s are equal on the primes and 1 then they are equal everywhere.equal on

Reply:
@eric-wieser is this still relevant after your commits?

Reply:
I don't think so